File: (39)sameNodeInTwoList/Solution.java
class ListNode {
    int val;
    ListNode next = null;

    ListNode(int val) {
        this.val = val;
    }
}
public class Solution {
	//蛮力法
    public static ListNode FindFirstCommonNode(ListNode pHead1, ListNode pHead2) {
    	if(pHead1==null||pHead2==null)
    		return null;
    	ListNode node1 = pHead1;
    	ListNode node2 = pHead2;
    	while(node1!=null){
    		node2 = pHead2;
    		while(node2!=null){
    			if(node2.val == node1.val)
    				return node2;
    			node2 = node2.next;
    		}
    		node1 = node1.next;
    	}
    	return null;
    	
}

    public static void main (String [] args){
    	ListNode node1 = new ListNode(1);
    	ListNode node2 = new ListNode(2);
    	node1.next = node2;
    	ListNode node3 = new ListNode(2);
    	System.out.println(FindFirstCommonNode(node1,node3).val);
    }
}

File: (39)sameNodeInTwoList/Solution.cpp
/*
struct ListNode {
	int val;
	struct ListNode *next;
	ListNode(int x) :
			val(x), next(NULL) {
	}
};*/
class Solution {
public:
    ListNode* FindFirstCommonNode( ListNode* pHead1, ListNode* pHead2) {
       if(pHead1 == NULL || pHead2 == NULL)
       		return NULL;
       	
       	int length1 = findListLength(pHead1);
       	int length2 = findListLength(pHead2);

       	int margin = 0;
       	margin = length1 - length2;
       	if(length1 > length2) {
       		return findSameNode(pHead1, pHead2, margin);
       	} else {
       		return findSameNode(pHead2, pHead1, -margin);
       	}
       	return NULL;
    }
 
    int findListLength(ListNode * node) {
    	ListNode *p = node;
    	int length = 0;
    	if(p != NULL) {
    		length++;
    		p = p->next;
    	}
    	return length;
    }

    ListNode* findSameNode(ListNode *node1, ListNode * node2, int margin) {
    	ListNode *p1 = node1;
    	ListNode *p2 = node2;
    	while(p1 != NULL && p2 !=NULL) {
    		if(p1 == p2)
    			return p1;
    		if(margin > 0) {
    			margin --;
    		} else {
    			p2 = p2->next;
    		}
    		p1 = p1->next;
    	}
    }

};